src/main/java/games/strategy/engine/lobby/server/userDB/DbUserController.java

```diff
@@ -18,39 +19,57 @@
  *  This is a migration tool, to go from one data source to another. When we switch over completely
  *  we can simplify and write/read to the primary datasource directly.
  */
+// TODO: rename to: UserController
 public class DbUserController implements UserDao {
   private static final Logger logger = Logger.getLogger(DbUserController.class.getName());
 
-  private static final Collection<UserDaoPrimarySecondary> implementations = Arrays.asList(
-      new DerbyUserController(),
-      new SqlUserController()
-  );
-
   /**
    * Stores a convenience reference to the authoritative DAO implementation.
    * We use this datasource for reading, we always write to it, any errors during read/write are considered
    * critical (on the other hand secondary DAO read/write errors can be logged and ignored)
    */
-  private static final UserDao primaryDao = determinePrimaryDataSource();
+  private final UserDao primaryDao;
+  private final Collection<UserDao> secondaryDaoSet;
+
+
+  /**
+   * The default constructor uses a production configuration.
+   */
+  public DbUserController() {
+    this(
+        new DerbyUserController(
+            UserDaoPrimarySecondary.Role.PRIMARY,
+            () -> Database.getDerbyConnection()),
```

Comment:
Nitpicking... these `Supplier`s can be replaced with method references, e.g. `Database::getDerbyConnection` and `Database::getPostgresConnection`.